Flash Chat iOS13/Controllers/ThreadListViewController.swift

```diff
@@ -0,0 +1,129 @@
+//
+//  ThreadListViewController.swift
+//  Flash Chat
+//
+//  Created by TechWithTyler on 10/13/22.
+//  Copyright © 2022 Angela Yu. All rights reserved.
+//
+
+import UIKit
+import Firebase
+
+class ThreadListViewController: UITableViewController {
+
+	var threads: [Thread] = []
+
+	let database = Firestore.firestore()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+		loadThreads()
+    }
+
+	func loadThreads() {
+		database.collection(Constants.FStore.threadsCollectionName)
+			.order(by: Constants.FStore.dateField, descending: false)
+			.addSnapshotListener { [self] (querySnapshot, error) in
+				threads = []
+				guard Auth.auth().currentUser != nil else { return }
+				if let error = error {
+					AppDelegate.showError(error, inViewController: self)
+				} else {
+					if let snapshotDocuments = querySnapshot?.documents {
+						for doc in snapshotDocuments {
+							let data = doc.data()
+							if let recipients = data[Constants.FStore.recipientsField] as? [String],
+							   let bubbles = data[Constants.FStore.bubblesField] as? [Message], let idString = data[Constants.FStore.idField] as? String {
+								let newThread = Thread(idString: idString, recipients: recipients, messageBubbles: bubbles)
+								threads.append(newThread)
+								DispatchQueue.main.async { [self] in
+									let indexPath = IndexPath(row: threads.count - 1, section: 0)
+									tableView?.reloadData()
+									tableView?.scrollToRow(at: indexPath, at: .top, animated: true)
+								}
```

Comment:
⛏️ 
^^ (emoji for a "nitpick" comment; optional to respond to)

These nested `if` statements are what's known as a "code smell"; something to recognize as an opportunity for a possible refactor for clarity / readability.

Wondering if you see an opportunity here to break any of this into a separate function, so that the code isn't indented as far to the right.